src/modules/data/blocks/ticket/sagas.js

```diff
@@ -737,6 +739,146 @@ export function* setTicketTempDetails( action ) {
+ * @borrows TEC - Functionality requires TEC to be enabled
+ * @param {String} prevStartDate Previous start date before latest set date time changes
+ * @export
+ */
+export function* syncTicketSaleEndWithEventStart( prevStartDate, blockId ){
+	try {
+		const tempEndMoment = yield select( selectors.getTicketTempEndDateMoment, { blockId } );
+		const endMoment = yield select( selectors.getTicketEndDateMoment, { blockId } );
+		const { moment: prevEventStartMoment } = yield call( createDates, prevStartDate );
+
+		// NOTE: Mutation
+		// Convert to use local timezone
+		yield all( [
+			call( [ tempEndMoment, 'local' ] ),
+			call( [ endMoment, 'local' ] ),
+			call( [ prevEventStartMoment, 'local' ] ),
+		] );
+
+		// If initial end and current end are the same, the RSVP has not been modified
+		const isNotManuallyEdited = yield call( [ tempEndMoment, 'isSame' ], endMoment, 'minute' );
+		const isSyncedToEventStart = yield call( [ tempEndMoment, 'isSame' ], prevEventStartMoment, 'minute' );
+
+		if ( isNotManuallyEdited && isSyncedToEventStart ) {
+			const eventStart = yield select( window.tribe.events.data.blocks.datetime.selectors.getStart );
+			const {
+				moment: endDateMoment,
+				date: endDate,
+				dateInput: endDateInput,
+				time: endTime,
+				timeInput: endTimeInput,
+			} = yield call( createDates, eventStart );
+
+			yield all( [
+				put( actions.setTicketTempEndDate( blockId, endDate ) ),
+				put( actions.setTicketTempEndDateInput( blockId, endDateInput ) ),
+				put( actions.setTicketTempEndDateMoment( blockId, endDateMoment ) ),
+				put( actions.setTicketTempEndTime( blockId, endTime ) ),
+				put( actions.setTicketTempEndTimeInput( blockId, endTimeInput ) ),
+
+				// Sync Ticket end items as well so as not to make state 'manually edited'
+				put( actions.setTicketEndDate( blockId, endDate ) ),
+				put( actions.setTicketEndDateInput( blockId, endDateInput ) ),
+				put( actions.setTicketEndDateMoment( blockId, endDateMoment ) ),
+				put( actions.setTicketEndTime( blockId, endTime ) ),
+				put( actions.setTicketEndTimeInput( blockId, endTimeInput ) ),
+
+				// Trigger UI button
+				put( actions.setTicketHasChanges( blockId, true ) ),
+			] );
+
+			yield fork( saveTicketWithPostSave, blockId );
+		}
+	} catch ( error ) {
+		// ¯\_(ツ)_/¯
+		console.error( error );
+	}
+}
+
+/**
+ * Listens for event start date and time changes after RSVP block is loaded.
+ * @borrows TEC - Functionality requires TEC to be enabled and post type to be event
+ * @export
+ */
+export function* handleEventStartDateChanges() {
+	try {
+		// Ensure we have a postType set before proceeding
+		const postTypeChannel = yield call( hasPostTypeChannel );
+		yield take( postTypeChannel );
+		yield call( [ postTypeChannel, 'close' ] );
+
+		const isEvent = yield call( isTribeEventPostType );
+		if ( isEvent && window.tribe.events ) {
+			const { SET_START_DATE_TIME, SET_START_TIME } = window.tribe.events.data.blocks.datetime.types;
```

Comment:
Looks good quick question about `window.tribe.events.data.blocks.datetime.types` as it might not be defined on post that are not events that has support for Tickets like `Post` or `Pages` or others.

Reply:
Yup it shouldn't work on those types, only on events

Reply:
Great sorry my question was if this code is executed there so in that case will trigger an error because `window.tribe.events.data.blocks.datetime` is not there.

Reply:
it's in a `try` block and there's a `window.tribe.events` check as well

Reply:
are you saying to throw an error?

Reply:
Oh that's right I overlooked that one 👍 thanks for answering my silly questions 💃 🕺